internal/pkg/scaffold/olm-catalog/descriptor.go

```diff
@@ -0,0 +1,216 @@
+	scanner := bufio.NewScanner(strings.NewReader(comments))
+	for scanner.Scan() {
+		l := strings.TrimSpace(scanner.Text())
+		if l == "" || strings.Contains(l, "+k8s:") || strings.Contains(l, "+kubebuilder:") {
+			continue
+		}
+		lines = append(lines, l)
+	}
+	if err := scanner.Err(); err != nil {
+		log.Error(err)
+	}
+	return strings.Join(lines, " ")
+}
+
+var mapRe = regexp.MustCompile(`map\[.+\]`)
+
+// processType cleans and returns the string of a type expression within fset.
+func processType(fset *token.FileSet, e ast.Expr) (t string) {
+	tbuf := &bytes.Buffer{}
+	if err := printer.Fprint(tbuf, fset, e); err != nil {
+		log.Fatal(err)
+	}
+	t = tbuf.String()
+	// Clean slice, map, and pointer syntax.
+	tt := strings.Replace(t, "[]", "", -1)
+	tt = strings.Replace(tt, "*", "", -1)
+	tt = mapRe.ReplaceAllString(tt, "")
+	// Only return non-primitive types without extra syntax.
+	if types.Universe.Lookup(tt) == nil {
+		t = tt
+	}
+	return t
+}
+
+// From https://github.com/openshift/console/blob/master/frontend/public/components/operator-lifecycle-manager/descriptors/types.ts#L5-L14
+var specXDescriptors = map[string][]string{
+	"size":                 {"size", "urn:alm:descriptor:com.tectonic.ui:podCount"},
+	"endpoints":            {"endpointList", "urn:alm:descriptor:com.tectonic.ui:endpointList"},
+	"label":                {"label", "urn:alm:descriptor:com.tectonic.ui:label"},
+	"resourceRequirements": {"resourceRequirements", "urn:alm:descriptor:com.tectonic.ui:resourceRequirements"},
+	"selector":             {"selector", "urn:alm:descriptor:com.tectonic.ui:selector:"},
+	"namespaceSelector":    {"namespaceSelector", "urn:alm:descriptor:com.tectonic.ui:namespaceSelector"},
+	"booleanSwitch":        {"booleanSwitch", "urn:alm:descriptor:com.tectonic.ui:booleanSwitch"},
+}
+
+// From https://github.com/openshift/console/blob/master/frontend/public/components/operator-lifecycle-manager/descriptors/types.ts#L16-L27
+var statusXDescriptors = map[string][]string{
+	"size":               {"size", "urn:alm:descriptor:com.tectonic.ui:podCount"},
+	"podStatuses":        {"podStatuses", "urn:alm:descriptor:com.tectonic.ui:podStatuses"},
+	"links":              {"w3Link", "urn:alm:descriptor:org.w3:link"},
+	"conditions":         {"conditions", "urn:alm:descriptor:io.kubernetes.conditions"},
+	"text":               {"text", "urn:alm:descriptor:text"},
+	"prometheusEndpoint": {"prometheusEndpoint", "urn:alm:descriptor:prometheusEndpoint"},
+	"status":             {"phase", "urn:alm:descriptor:io.kubernetes.phase"},
+	"reason":             {"reason", "urn:alm:descriptor:io.kubernetes.phase:reason"},
+}
+
+var jsonTagRe = regexp.MustCompile("`json:\"([^,]+),?.*\"`")
+
+// guessPathAndXDescriptorFromTag uses json field tags to guess which path
+// and x-descriptor a CRD should have. This choice is a guess based on tag
+// identifier.
+func guessPathAndXDescriptorFromTag(tag string, isSpec bool) (path string, xd []string) {
```

Comment:
Why are we guessing the path instead of using the type hierarchy from `_types.go`?

Reply:
You mean you'd rather have `path` derived from type names themselves? Here I'm using JSON tags because they follow Kubernetes resource serialization conventions, whereas type names are arbitrary.

Reply:
`path` is what other components use to find the value associated with the x-descriptor (right?) so it definitely makes sense to use the JSON tag. The part I'm confused about is the lookup from tag to path, where that path can be different than the tag. (e.g. `links` -> `w3link`). 

Shouldn't `path` be built using the tags directly instead of doing a lookup?